### ISSUE TITLE
Image preview on SpatieMediaLibraryFileUpload component doesn't show stored image name

### ISSUE THREAD

**Issue by mikemathewson** [laravel-filament/filament#7530]
### Package

filament/filament

### Package Version

v3.0.8

### Laravel Version

v9.52.14

### Livewire Version

v3.0.0-beta.6

### PHP Version

PHP 8.1.0

### Problem description

When using the SpatieMediaLibraryFileUpload component, the image preview doesn't display the original filename in the preview (after saving). A workaround is to use preserveFilenames() however the original image name is stored in the database.

[media-bug-screen-capture.webm](https://github.com/filamentphp/filament/assets/22248626/44ecbe77-1678-4fbf-ad07-a6186ad4f00a)

This issue seems to be due to how the FilePond Image Preview plugin works. If any file data is passed into the plugin then it doesn't generate the preview. This is probably why Filament's file-upload.js doesn't pass in the file data for image types:

```
files.push({
    source: uploadedFile.url,
    options: {
        type: 'local',
        ...(/^image/.test(uploadedFile.type)
            ? {}
            : {
                  file: {
                      name: uploadedFile.name,
                      size: uploadedFile.size,
                      type: uploadedFile.type,
                  },
              }),
    },
})
```

A potential solution is to use the FilePond [Poster Plugin](https://pqina.nl/filepond/docs/api/plugins/file-poster/) instead of the Image Preview plugin and update the file-upload.js:

```
files.push({
    source: uploadedFile.url,
    options: {
        type: 'local',
        file: {
            name: uploadedFile.name,
            size: uploadedFile.size,
            type: uploadedFile.type,
        },
        ...(/^image/.test(uploadedFile.type)
            ? {
                metadata: {
                  poster: uploadedFile.poster || uploadedFile.url
                },
            }
            : {}),
    },
})
```

This would also have the benefit of displaying the file information sooner while the preview is loading.

### Expected behavior

The original filename which is stored in the database should be displayed in the image preview

### Steps to reproduce

1. Add the SpatieMediaLibraryFileUpload component to a resource
2. Upload an image (NOTE: correct name displays in the image preview)
3. Save record
4. Image preview now shows the randomized filename

### Reproduction repository

https://github.com/mikemathewson/filament-media-bug

**Comment by danharrin** [comment]
Please submit a PR if you can

**Comment by danharrin** [comment]
Hi, I've just looked into it. You're correct, we cant put the file name in the header of the image as it will conflict with the preview plugin.

I've just tried replacing it with the poster plugin. We can not use it, because it does not support previewing newly-uploaded files, just existing ones.

So we are kinda stuck. I think the best path forward is to submit a bug report to Filepond about the fact we cant specify a custom file name for image previews.

Whoever wants to open that issue, do it. Once you've done it, submit a PR to Filament to remove that condition. Or @ me and I will submit the Filament PR.
